main.tf

```
@@ -4,12 +4,15 @@ locals {
 
   db_rules = "${null_resource.subnet_mappings.*.triggers}"
 
-  vaultName = "infra-vault-${var.subscription}"
+  vaultName = var.key_vault_name != "" ? var.key_vault_name : "infra-vault-${var.subscription}"
+  resource_group_name = var.key_vault_rg != "" ? var.key_vault_rg : (
```

should specify that this is for vault
```suggestion
  vault_resource_group_name = var.key_vault_rg != "" ? var.key_vault_rg : (
```